src/libstd/sys/unix/process.rs

```diff
@@ -224,7 +223,8 @@ impl Process {
                 if !setup(err_fd, libc::STDERR_FILENO) { fail(&mut output) }
 
                 // close all other fds
-                for fd in (3..getdtablesize()).rev() {
+                let open_max = libc::sysconf(libc::consts::os::sysconf::_SC_OPEN_MAX);
```

Comment:
According to http://stackoverflow.com/a/918469 it sounds like some systems may return `INT_MAX` for this value. How come this changed from `getdtablesize`?